### PR TITLE
Restore Cursor after ctrl+c when prompt to download snapshot is active

### PR DESCRIPTION
**Pull request by jdjaustin**
**Summary of changes**
Changes introduced in this pull request:
- Cursor disappears after receiving ctrl+c interrupt when prompt to download snapshot is active. This PR aims to resolve this errant behavior and restore the cursor.

**Reference issue to close (if applicable)**

Closes issue #2204 

**Other information and links**

N/A

### CODE REVIEW

**Comment by jdjaustin** [comment]
`cargo clippy` was not happy about using `.unwrap()` instead of `?`, but the compiler doesn't allow the use of `?` inside of a closure that returns `()`, and modifying the closure to return a `Result` is not feasible because of the trait bounds defined in the `ctrlc` crate. If I'm missing an alternate method to resolve this, please let me know.

File: forest/daemon/src/daemon.rs [hunk 1]
```diff
@@ -427,15 +427,22 @@ async fn prompt_snapshot_or_die(config: &Config) -> bool {
...
+            if result {
+                true
+            } else {
+                cli_error_and_die("Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file] or --download-snapshot to download one automatically", 1);
+            }
+        }
+        Err(_) => {
+            std::mem::drop(should_download);
+            cli_error_and_die("Error getting input.", 1);
+        }
```
Comment (lemmih) [inline forest/daemon/src/daemon.rs:445]: Are these changes related to the cursor fix? If so, could you explain it in a comment. If not then you can delete this code.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: forest/daemon/src/daemon.rs (hunk 1 above)
Comment (jdjaustin) [reply forest/daemon/src/daemon.rs:445]: Yes, this was originally simply part of my troubleshooting, but I just tested it without the match statement, and it is required to properly handle the prompt resuts including the case of an interrupt. I'll add a comment.

**Review by lemmih — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]
